tooling/tracer/src/tail_diff_vecs.rs

```diff
@@ -0,0 +1,42 @@
+use std::cmp::min;
+
+/// Produce the "tail difference" of two vectors.
+///
+/// Here, this means that the common prefix of the two vectors is computed. The length of that
+/// prefix is returned, together with the suffix of the first vector that follows the common
+/// prefix, as well as the suffix of the second vector that follows the common prefix.
+///
+/// Example:
+/// ```
+/// use noir_tracer::tail_diff_vecs::tail_diff_vecs;
+/// assert_eq!(tail_diff_vecs(&vec!(1, 2, 3), &vec!(1, 2, 5, 6)), (2, vec!(3), vec!(5, 6)));
+///
+/// assert_eq!(tail_diff_vecs(&vec!(1, 2, 3), &vec!(5, 6)), (0, vec!(1, 2, 3), vec!(5, 6)));
+///
+/// assert_eq!(tail_diff_vecs(&vec!(1, 2, 3), &vec!(1, 2)), (2, vec!(3), vec!()));
+///
+/// assert_eq!(tail_diff_vecs(&vec!(1, 2, 3, 4), &vec!(1, 2, 5)), (2, vec!(3, 4), vec!(5)));
+/// ```
+pub fn tail_diff_vecs<T>(a: &Vec<T>, b: &Vec<T>) -> (usize, Vec<T>, Vec<T>)
+where
+    T: PartialEq + Clone,
+{
+    // Here's a functional version of this code, for clarity:
+    // let first_nomatch: usize = xs.iter().zip(ys.iter())
```

Comment:
You can short-circuit when first mismatch is found by using the position method of Iter.
```
    let first_nomatch: Option<usize> = a.iter().zip(b.iter())
        .position(|(x, y)| x != y);
```

Reply:
Thanks, final version is based on your comment.